### PR TITLE
Add diagrams for request flow. Clean up overstuffed language

### LINKED ISSUE #132: High-level overview of how a request gets processed
Back when I learned Rails, I remember that diagrams like these really helped me understand how the different components worked together when processing a request:

MVC diagram 

Diagram on how Rails processes a request  

[This section](https://luckyframework.org/guides/getting-started/concepts#file-structure-overview) lists the file structure but, since it's a long list, I find difficult to "put the pieces together" unless I re-read it several times.

### PR DESCRIPTION
**Pull request by paulcsmith**
Closes #132 

Would love thoughts on the diagram. I almost wonder if the text below is just as useful and the diagram is redundant? The text is also linkable and easier to update. So maybe we shouldn't do the diagram? I'm not really sure. Maybe I can do something to make the diagram more useful?

Screen Shot 2019-09-17 at 11 55 08 AM

### CODE REVIEW

File: src/actions/guides/getting-started/concepts.cr [hunk 1]
```diff
@@ -11,26 +11,34 @@ class Guides::GettingStarted::Concepts < GuideAction
...
-    * Minimize guesswork with conventions for the most common types of objects.
-    * Help developers break things into discrete pieces so things are easy to share
-      and maintain in the future.
-    * Minimize boilerplate code so it's easy to focus on what your app does better
+    * **Catch bugs at compile time**, rather than finding them in production.
+    * **Spend less time writing tests**, because the compiler catches many errors for you.
+    * Minimize guesswork with **conventions for common tasks**.
+    * Help developers break things into discrete pieces so things are **easy to share
+      and maintain as the codebase grows**.
+    * **Minimize boilerplate** code so it's easy to focus on what your app does better
```
Comment (paulcsmith) [inline src/actions/guides/getting-started/concepts.cr:19]: While I was in here I deleted a lot of the stuff and bolded the important bits.

**Review by jwoertink — APPROVED** [review]
Wow! This is amazing. This is going to help a ton of people.

**Comment by paulcsmith** [comment]
Do you like the diagram along with the text? Anything I can do to make it make more sense?

**Comment by jwoertink** [comment]
We should probably ask @raul, but I think it's ok.

**Comment by edwardloveall** [comment]
If you're willing to wait, I'd love to make a slightly better graph, but we could also merge and come back to it.

**Comment by paulcsmith** [comment]
I could also merge without the graph but keep the text, and then you can add a graph later? What do you think @edwardloveall?

**Comment by edwardloveall** [comment]
I think you should just merge it and I'll come back if I get a chance. This graph is better than no graph. And who knows, maybe it's the best graph! 😄

**Comment by paulcsmith** [comment]
@edwardloveall Sounds good! 

Also, I added a rounded white bg with a shadow
Screen Shot 2019-09-17 at 1 14 32 PM

**Comment by edwardloveall** [comment]
Yeah, that helps readability!

**Comment by raul** [comment]
I like the diagram and love the text - super useful intro, thanks a ton @paulcsmith! 🙌

**Pull request merged**